tests/unit-tests/test_model_save_load.py

```diff
@@ -0,0 +1,199 @@
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        g, _ = generate_dummy_dist_graph(tmpdirname)
+        feat_size = {"n0":0, "n1":0}
+        embed_layer = GSNodeEncoderInputLayer(g, feat_size, 32)
+
+        model_path = os.path.join(tmpdirname, "model")
+        os.mkdir(model_path)
+        sparse_embs = \
+            {ntype: sparse_emb._tensor[th.arange(embed_layer.g.number_of_nodes(ntype))] \
+                for ntype, sparse_emb in embed_layer.sparse_embeds.items()}
+        for i in range(world_size):
+            save_sparse_embeds(model_path, embed_layer, i, world_size)
+
+        for ntype in embed_layer.sparse_embeds.keys():
+            saved_embs = []
+            for i in range(world_size):
+                saved_embs.append(th.load(
+                    os.path.join(os.path.join(model_path, ntype),
+                                 f'sparse_emb_{i}.pt')))
+            saved_embs = th.cat(saved_embs, dim=0)
+            assert_equal(saved_embs.numpy(), sparse_embs[ntype].numpy())
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+@pytest.mark.parametrize("infer_world_size", [3, 8, 16])
+@pytest.mark.parametrize("train_world_size", [8])
+def test_sparse_embed_load(infer_world_size, train_world_size):
+    """ Test sparse embedding loading logic. (graphstorm.model.utils.load_sparse_embeds)
+
+        It uses save_sparse_embeds to save the modal and loads it using
+        load_sparse_embeds with different infer_world_size to mimic
+        different number of processes to load the sparse embedding.
+        It will compare the embedings stored and loaded.
+    """
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        g, _ = generate_dummy_dist_graph(tmpdirname)
+        feat_size = {"n0":0, "n1":0}
+        embed_layer = GSNodeEncoderInputLayer(g, feat_size, 4)
+        model_path = os.path.join(tmpdirname, "model")
+        os.mkdir(model_path)
+        saved_embs = \
+            {ntype: sparse_emb._tensor[th.arange(embed_layer.g.number_of_nodes(ntype))] \
+                for ntype, sparse_emb in embed_layer.sparse_embeds.items()}
+        #
+        for i in range(train_world_size):
+            save_sparse_embeds(model_path, embed_layer, i, train_world_size)
+
+        for i in range(infer_world_size):
+            load_sparse_embeds(model_path, embed_layer, i, infer_world_size)
```

Comment:
it's not clear to me whether you have tested the following three cases:

# 1. N == K
# 2. N > K, some trainers/infers need to load more than one files
# 3. N < K, some trainers/infers do not need to load any files

Reply:
We have 
`
@pytest.mark.parametrize("infer_world_size", [3, 8, 16])
@pytest.mark.parametrize("train_world_size", [8])
`
So the combinations will be 
N = 8  and K = 3
N = 8  and K = 8
N = 8  and K = 16
To test different cases.